app/clubDetails.tsx

```diff
@@ -0,0 +1,136 @@
+import { View, StyleSheet, Text, ScrollView, ActivityIndicator } from 'react-native';
+import React from 'react';
+import { useNavigation } from '@react-navigation/native';
+import { useLocalSearchParams } from 'expo-router';
+import { useQuery } from '@tanstack/react-query';
+import { fetchClubInfo } from '../api/clubService';
+import { ClubInfo as ClubInfoType } from '../types/ClubInfoTypes';
+import { VStack } from '@/components/ui/vstack';
+import CardLayout from '@/components/cards/CardLayout';
+
+export default function ClubDetails() {
+  const navigation = useNavigation();
+  navigation.setOptions({ title: `Club Info` });
+  const params = useLocalSearchParams();
+  const clubId = Array.isArray(params.id) ? params.id[0] : params.id || '';
+  
+  const { data: clubInfo, isLoading, error } = useQuery<ClubInfoType, Error>(
+    {
+      queryKey: ['clubInfo', clubId],
+      queryFn: () => fetchClubInfo(clubId),
+      enabled: !!clubId,
+    }
+  );
+
+  if (isLoading) {
+    return <ActivityIndicator size="large" color="#0000ff" />;
+  }
+
+  if (error) {
+    return (
+      <View style={styles.errorContainer}>
+        <Text style={styles.errorText}>Data unavailable.</Text>
+      </View>
+    );
+  }
+
+  return (
+    <View style={styles.container}>
+      <ScrollView contentContainerStyle={styles.scrollContainer} showsVerticalScrollIndicator={false}>
+        {clubInfo && 
+          <>
+            <Text style={styles.text}>{clubInfo.name} stats</Text>
+            <VStack space="lg" style={styles.vStackContainer}>
+              <CardLayout>
```

Comment:
maybe you could do something similar to PlayersByPosition here? if you can get a label you could even group the stats and do:

```
stats.map((stat, i) => (
      <View key={i} style={styles.row}>
        <Text style={styles.playerDetails}>{stat.label}:</Text>
        <Text style={styles.playerStat}>{stat.value}</Text>
      </View>
    ))
```
but constructing a label from the keys you get in clubInfo seems a pain in the ass 😅